File: frontend/src/types/collection.ts
export type { Result, ResultError } from './common'

export interface Collection {
  id: string
  workspaceId: string
  parentId: string | null
  name: string
  description: string
  authType: string
  authData: string
  preScript: string
  postScript: string
  sortOrder: number
  version: number
  createdAt: string
  updatedAt: string
}

export interface CollectionTreeNode extends Collection {
  children: CollectionTreeNode[]
}

File: internal/domain/entities/collection.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type Collection struct {
	ID           uuid.UUID
	WorkspaceID  uuid.UUID
	ParentID     *uuid.UUID
	Name         string
	PreScript    string
	PostScript   string
	Description  string
	AuthType     AuthType
	AuthData     string
	GRPCMetadata []HeaderItem
	SortOrder    int
	Version      int
	IsDelete     bool
	CreatedBy    string
	CreatedAt    time.Time
	UpdatedBy    string
	UpdatedAt    time.Time
}
